app/feature/feature-payments/src/main/kotlin/com/hedvig/android/feature/payments/discounts/DiscountsDestination.kt

```diff
@@ -146,31 +151,168 @@ internal fun DiscountsScreen(
           color = MaterialTheme.colorScheme.onSurfaceVariant,
         )
       } else {
-        discounts.forEachIndexed { index, discount ->
-          DiscountRow(discount)
-          if (index < discounts.size - 1) {
-            HorizontalDivider()
-          }
-        }
+        Spacer(modifier = Modifier.height(16.dp))
+        DiscountRows(discounts)
       }
 
       Spacer(modifier = Modifier.height(16.dp))
       HedvigSecondaryContainedButton(
         text = stringResource(id = R.string.PAYMENTS_ADD_CAMPAIGN_CODE),
         onClick = { onShowBottomSheet() },
       )
+      if (uiState.foreverInformation != null) {
+        Spacer(modifier = Modifier.height(32.dp))
+        ForeverSection(uiState.foreverInformation, Modifier)
+      }
+      Spacer(modifier = Modifier.height(16.dp))
+      Spacer(
+        modifier = Modifier.padding(WindowInsets.safeDrawing.only(WindowInsetsSides.Bottom).asPaddingValues()),
+      )
+    }
+  }
+}
+
+@Composable
+private fun ForeverSection(foreverInformation: ForeverInformation, modifier: Modifier = Modifier) {
+  Column(modifier) {
+    val incentive = foreverInformation.potentialDiscountAmountPerNewReferral.toString()
+    var showForeverInfoBottomSheet by remember { mutableStateOf(false) }
+    if (showForeverInfoBottomSheet) {
+      HedvigInfoBottomSheet(
+        onDismissed = { showForeverInfoBottomSheet = false },
+        title = stringResource(R.string.referrals_info_sheet_headline),
+        body = stringResource(R.string.referrals_info_sheet_body, incentive),
+      )
     }
+    HorizontalItemsWithMaximumSpaceTaken(
+      startSlot = {
+        Text(
+          stringResource(R.string.PAYMENTS_REFERRALS_INFO_TITLE),
+          modifier = Modifier.wrapContentSize(Alignment.CenterStart),
+        )
+      },
+      endSlot = {
+        Icon(
+          imageVector = Icons.Hedvig.InfoFilled,
+          tint = MaterialTheme.colorScheme.onSurfaceVariant,
+          contentDescription = null,
+          modifier = Modifier
+            .wrapContentSize(Alignment.CenterEnd)
+            .size(16.dp)
+            .clip(MaterialTheme.shapes.squircleLarge)
+            .clickable { showForeverInfoBottomSheet = true }
+            .minimumInteractiveComponentSize(),
+        )
+      },
+    )
+    Spacer(modifier = Modifier.height(16.dp))
+    HorizontalItemsWithMaximumSpaceTaken(
+      startSlot = {
+        HedvigCard(
```

Comment:
Maybe also add copying the code on click here?

Reply:
Will do!